### PR TITLE
Advantagekit implementation v2

### PR DESCRIPTION
**Pull request by arghunter**
Advantage kit works with minor changes and logs most of what we need.

### CODE REVIEW

**Comment by arghunter** [comment]
One thing to change  later maybe logging to a usb instead of directly to the rio.

File: RobotBase.java [hunk 1]
```diff
@@ -54,6 +68,21 @@ public RobotContainerBase getRobotContainer() {
    */
   @Override
   public void robotInit() {
+    Logger.getInstance().recordMetadata("ProjectName", "2023-Robot"); // Set a metadata value
+
+    if (isReal()) {
+        Logger.getInstance().addDataReceiver(new WPILOGWriter("/home/lvuser")); // Log to a USB stick
```
Comment (adias408) [inline RobotBase.java]: do we not have a usb drive?

File: RobotBase.java [hunk 2]
```diff
@@ -54,6 +68,21 @@ public RobotContainerBase getRobotContainer() {
...
+        Logger.getInstance().addDataReceiver(new NT4Publisher()); // Publish data to NetworkTables
```
Comment (adias408) [inline RobotBase.java]: we should keep an eye on how much overhead this adds. we may not want it in matches

File: RobotBase.java [hunk 3]
```diff
@@ -117,20 +146,18 @@ public static synchronized RobotBase getInstance() {
...
-   */
+
   @Override
   public void disabledPeriodic() {
     robotContainer.disabledPeriodic();
   }
 
   /**
-   * This autonomous runs the autonomous command selected by Sebby
+   * This autonomous runs the autonomous command selected by your
```
Comment (adias408) [inline RobotBase.java]: nit: This runs the selected autonomous command (or something)

File: RobotBase.java [hunk 4]
```diff
@@ -117,20 +146,18 @@ public static synchronized RobotBase getInstance() {
...
    */
   @Override
   public void autonomousInit() {
-    Logger.consoleLog("Autonomous Init");
+    // Logger.consoleLog("Autonomous Init");
```
Comment (adias408) [inline RobotBase.java]: why not just delete these lines rather than commenting them out?

File: subsystems/drivebase/SwerveDrive.java [hunk 5]
```diff
@@ -225,8 +228,13 @@ public void mustangPeriodic() {
             }
         }
         mPoseEstimator.update();
-        SmartDashboard.putNumber("navX heading", getPose().getRotation().getDegrees());
-        SmartDashboard.putNumber("pitch", getPitch());
+        Logger.getInstance().recordOutput(getName()+"/navX Heading Deg", getPose().getRotation().getDegrees());
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java]: since this is library code and it's going to run a lot: java strings aren't the cheapest thing to build. Here we've got 3 strings (the name, the property name and then the concatenation of the two). In addition to the cycles to create those, we're generating more garbage to be collected. Could we not just create constants that we use here?

File: subsystems/drivebase/SwerveDrive.java [hunk 6]
```diff
@@ -225,8 +228,13 @@ public void mustangPeriodic() {
...
+        Logger.getInstance().recordOutput(getName()+"/pitch",getPitch());
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java]: a style thing: there should be a space after the comma. You can set up your editor to just do this for you. We're being inconsistent about this within two lines (see previous line)

File: subsystems/drivebase/SwerveDrive.java [hunk 7]
```diff
@@ -225,8 +228,13 @@ public void mustangPeriodic() {
...
+        Logger.getInstance().recordOutput(getName()+"/roll",getRoll());
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java]: why do we call Logger.getInstance() repeatedly rather than just once and reusing the variable? Is there a need to call it once per periodic even? Could we stash it in the object instance? Or even as a class variable that we use everywhere effectively as a global? Depending how it works, one or more of those options might not be possible (if it's doing something custom per invocation or per object) but at minimum I expect we can reuse it within this method.

Related: is there a version of recordOutput() that takes multiple key/value pairs to avoid making three calls?

Is there a return value that we should be looking at? Or even if there is, do we not care?

File: subsystems/drivebase/SwerveDrive.java [hunk 8]
```diff
@@ -1,11 +1,14 @@
...
 import java.util.Map;
+
+import org.littletonrobotics.junction.Logger;
+
 import com.pathplanner.lib.PathPlannerTrajectory;
 import com.pathplanner.lib.auto.SwerveAutoBuilder;
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.CANSparkMax.IdleMode;
-
+import frc.team670.mustanglib.subsystems.VisionSubsystemBase;
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java:12]: why was this added? I don't see it being used in any of the new code below. Did I miss it? If not, you can probably configure your editor to clean out unnecessary imports

File: utils/SwervePoseEstimatorBase.java [hunk 9]
```diff
@@ -20,7 +21,7 @@
 import edu.wpi.first.wpilibj.smartdashboard.Field2d;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
-import frc.team670.mustanglib.subsystems.VisionSubsystemBase;
+// import frc.team670.mustanglib.subsystems.VisionfSubsystemBase;
```
Comment (adias408) [inline utils/SwervePoseEstimatorBase.java:23]: Why is this here commented out (with a typo) and it's also added above on line 7?

File: utils/SwervePoseEstimatorBase.java [hunk 10]
```diff
@@ -112,8 +113,8 @@ public void update() {
 
         poseEstimator.update(driveBase.getGyroscopeRotation(), driveBase.getModulePositions());
         field2d.setRobotPose(getAbsoluteFieldOrientedPoseFromAllianceOriented());
-        SmartDashboard.putString("Estimated Pose",
-                getFormattedPose(getAbsoluteFieldOrientedPoseFromAllianceOriented()));
+        Logger.getInstance().recordOutput("DriveBase/Estimated Pose",
```
Comment (adias408) [inline utils/SwervePoseEstimatorBase.java]: Why are we hardcoding DriveBase here whereas above we were using getName()? Same comment as elsewhere: let's make this key a constant

**Review by adias408 — CHANGES_REQUESTED** [review]

File: RobotBase.java (hunk 4 above)
Comment (arghunter) [reply RobotBase.java]: I thought I deleted them, i'll get on that tomorrow

Comment (adias408) [reply RobotBase.java]: When you create a PR: look at the diff before you send it out. At least skim through it. You'll find something that you thought you'd done or forgot about or whatever very nearly every time. And if you go through it as if you were the reviewer, you'll probably find quite a few things you want to change before you send it out. Or maybe it's just me - I know I usually do. It's that self-editing step that your English teachers have been talking about since you learned to write. Same thing applies here.

Comment (arghunter) [reply RobotBase.java]: Yeah, I definitely created this PR hastily. Well, its just a lesson for the future.

File: subsystems/drivebase/SwerveDrive.java (hunk 7 above)
Comment (arghunter) [reply subsystems/drivebase/SwerveDrive.java]: There is no return value and it doesn't like there are any methods for pushing multiple key-value pairs  in one call. Also, Logger has become static
https://github.com/Mechanical-Advantage/AdvantageKit/blob/main/junction/core/src/org/littletonrobotics/junction/Logger.java

File: subsystems/drivebase/SwerveDrive.java (hunk 5 above)
Comment (arghunter) [reply subsystems/drivebase/SwerveDrive.java]: We switched to constants

File: RobotBase.java (hunk 1 above)
Comment (arghunter) [reply RobotBase.java]: not really, i just found a 120mb flash drive in the kit of parts. We probably want a drive that is  a bit bigger like 4 or 8 gb so that we don't overwrite logs accidentally

Honestly even 1 gb should be fine

File: commands/MustangCommand.java [hunk 11]
```diff
@@ -11,7 +11,7 @@
  * 
  * @author lakshbhambhani, ethan c, armaan, aditi
  */
-public interface MustangCommand extends Command {
+public interface MustangCommand {
```
Comment (adias408) [inline commands/MustangCommand.java]: we've already made these changes, right? you should merge the 2024 branch up into this branch. there may be some conflicts you need to resolve

File: commands/RamseteCommand.java [hunk 12]
```diff
@@ -162,9 +162,9 @@ public void execute() {
       rightOutput = rightSpeedSetpoint;
     }
 
-    // Logger.consoleLog("LeftspeedSetPoint: %s, Left Speed: %s, LeftOutput: %s, rightSpeedSetPoint: %s Right Speed: %s rightOutput: %s", leftSpeedSetpoint, m_speeds.get().leftMetersPerSecond, leftOutput, rightSpeedSetpoint, m_speeds.get().rightMetersPerSecond, rightOutput);
-    // Logger.consoleLog("Trajectory %s", m_trajectory.sample(curTime));
-    // Logger.consoleLog("Pose: %s", m_pose.get().toString());
+    ConsoleLogger.consoleLog("LeftspeedSetPoint: %s, Left Speed: %s, LeftOutput: %s, rightSpeedSetPoint: %s Right Speed: %s rightOutput: %s", leftSpeedSetpoint, m_speeds.get().leftMetersPerSecond, leftOutput, rightSpeedSetpoint, m_speeds.get().rightMetersPerSecond, rightOutput);
```
Comment (adias408) [inline commands/RamseteCommand.java:165]: do we really want to revive these? just seems like noise

separately: we don't use RamsetCommand anymore, do we?

File: commands/drive/teleop/swerve/XboxSwerveDrive.java [hunk 13]
```diff
@@ -35,7 +38,7 @@ public XboxSwerveDrive(SwerveDrive swerveDriveBase, MustangController controller
 
         MAX_VELOCITY = swerveDriveBase.getMaxVelocityMetersPerSecond();
         MAX_ANGULAR_VELOCITY = swerveDriveBase.getMaxAngularVelocityMetersPerSecond();
-
+        SmartDashboard.putNumber("Swerve Speed", 0.66);
```
Comment (adias408) [inline commands/drive/teleop/swerve/XboxSwerveDrive.java]: why is this added as part of this change? and also, what does it mean? 0.66?

File: commands/drive/teleop/swerve/XboxSwerveDrive.java [hunk 14]
```diff
@@ -58,8 +61,8 @@ public void execute() {
             thetaVel = rotPIDController.calculateRotationSpeed(driveBase.getGyroscopeRotation(),
                     desiredHeading);
         }
-
-        driveBase.drive(ChassisSpeeds.fromFieldRelativeSpeeds(xVel, yVel, thetaVel,
+        double speedMult= SmartDashboard.getNumber("Swerve Speed", 0.66);
```
Comment (adias408) [inline commands/drive/teleop/swerve/XboxSwerveDrive.java]: what is this?

File: commands/drive/teleop/tank/XboxRocketLeague/XboxCurvatureDrive.java [hunk 15]
```diff
@@ -37,7 +37,7 @@ public class XboxCurvatureDrive extends CommandBase implements MustangCommand {
 
     /**
      * Constructs a new XboxCurvatureDrive.
-     * @param driveBase the drivebase to be driven
+     * @param driveBase the drrivebase to be drivnr
```
Comment (adias408) [inline commands/drive/teleop/tank/XboxRocketLeague/XboxCurvatureDrive.java]: why are we adding typos? :)

File: commands/drive/teleop/tank/XboxRocketLeague/XboxCurvatureDrive.java [hunk 16]
```diff
@@ -46,7 +46,7 @@ public XboxCurvatureDrive(TankDrive driveBase, MustangController controller) {
         this.controller = controller;
         addRequirements(driveBase);
         healthRequirements.put(driveBase, HealthState.YELLOW);
- 
+
```
Comment (adias408) [inline commands/drive/teleop/tank/XboxRocketLeague/XboxCurvatureDrive.java]: ?

File: commands/drive/teleop/tank/XboxRocketLeague/XboxCurvatureDrive.java [hunk 17]
```diff
@@ -64,4 +64,5 @@ public Map<MustangSubsystemBase, HealthState> getHealthRequirements() {
         return healthRequirements;
     }
 
+
```
Comment (adias408) [inline commands/drive/teleop/tank/XboxRocketLeague/XboxCurvatureDrive.java]: ?

File: dataCollection/sensors/ColorMatcher.java [hunk 18]
```diff
@@ -63,6 +65,10 @@ private Color getTargetColor() {
 
   public ColorMatcher() {
     init();
+    COLORMATCHER_RED = "ColorMatcher/Red";
```
Comment (adias408) [inline dataCollection/sensors/ColorMatcher.java:68]: maybe as a separate task but we should have a port number or some identifier associated with these in case we have multiple color sensors

File: dataCollection/sensors/IRSensor.java [hunk 19]
```diff
@@ -31,10 +34,10 @@ public boolean isTriggered() {
     }
 
     /**
-     * Write the state of this IR sensor to SmartDashboard
+     * Write the state of this IR sensor to Advantage Scope
      */
     public void sendIRDataToDashboard() {
```
Comment (adias408) [inline dataCollection/sensors/IRSensor.java]: method name is wrong

File: dataCollection/sensors/NavX.java [hunk 20]
```diff
@@ -147,6 +147,8 @@ private synchronized void setOffSetAngle() {
         offSet = getRawYawDegrees();
     }
 
+
```
Comment (adias408) [inline dataCollection/sensors/NavX.java]: why is this file touched?

File: dataCollection/sensors/PicoColorMatcher.java [hunk 21]
```diff
@@ -60,6 +62,11 @@ private Color getTargetColor() {
 
   public PicoColorMatcher() {
     init();
+    COLORMATCHER_RED = "ColorMatcher/Red";
```
Comment (adias408) [inline dataCollection/sensors/PicoColorMatcher.java:65]: should have some kind of differentiator if we have multiple of these. or is that not possible with the current pico implementation? if not, then ok

File: dataCollection/sensors/TimeOfFlightSensor.java [hunk 22]
```diff
@@ -257,7 +258,7 @@ public int getAddress(){
      */
     protected void update() {
         // wait for device to be ready for range measurement
-        // Logger.consoleLog("Indexer read: %s");
+        ConsoleLogger.consoleLog("Indexer read: %s");
```
Comment (adias408) [inline dataCollection/sensors/TimeOfFlightSensor.java]: why do we want to revive this console logging?

File: subsystems/MustangSubsystemBase.java [hunk 23]
```diff
@@ -119,11 +122,12 @@ public final void periodic() {
     }
 
     public void pushHealthToDashboard() {
-        NetworkTableEntry subsystem = table.getEntry(this.getName());
-        subsystem.setString(getHealth(false).toString());
-        if (getHealth(false).toString().equals("YELLOW") || getHealth(false).toString().equals("RED")) {
-            // RobotContainer.notifyDriverController(1.0, 0.3);
-        }
+        Logger.recordOutput(BASE_HEALTH, getHealth(false).toString());
+        // NetworkTableEntry subsystem = table.getEntry(this.getName());
```
Comment (adias408) [inline subsystems/MustangSubsystemBase.java]: just delete these lines

File: subsystems/SparkMaxRotatingSubsystem.java [hunk 24]
```diff
@@ -212,11 +212,9 @@ public boolean setTemporaryTargetAngleInDegrees(double angle) {
      *         subsystem turning through this angle
      */
     protected double getMotorRotationsFromAngle(double angle) {
-        // double rotations = (angle / 360) * kConfig.kRotatorGearRatio
-        //         + ((int) (getUnadjustedPosition() / kConfig.kRotatorGearRatio))
-        //                 * kConfig.kRotatorGearRatio;
-        double rotations = ((angle - getCurrentAngleInDegrees())/360) * kConfig.kRotatorGearRatio + getUnadjustedPosition();
-
+        double rotations = (angle / 360) * kConfig.kRotatorGearRatio
```
Comment (adias408) [inline subsystems/SparkMaxRotatingSubsystem.java]: why are we changing this implementation in this PR?

File: subsystems/drivebase/SwerveDrive.java [hunk 25]
```diff
@@ -39,78 +43,103 @@ public abstract class SwerveDrive extends DriveBase {
...
 
         // back right
         mModules[3] = Mk4iSwerveModuleHelper.createNeo(
                 tab.getLayout("Back Right Module", BuiltInLayouts.kList).withSize(2, 4)
-                        .withPosition(6, 0),
-                kModuleConfig, config.kSwerveModuleGearRatio, config.kBackRightModuleDriveMotor,
-                config.kBackRightModuleSteerMotor, config.kBackRightModuleSteerEncoder,
-                config.kBackRightModuleSteerOffset);
+                    .withPosition(6, 0),
+            kModuleConfigBackRight, config.kSwerveModuleGearRatio, config.kBackRightModuleDriveMotor,
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java]: I'm confused why these changes are part of this PR. Aren't they already in the 2024 branch?

File: subsystems/drivebase/SwerveDrive.java [hunk 26]
```diff
@@ -254,23 +296,23 @@ public void setModuleStates(SwerveModuleState[] states) {
...
-        && Math.abs(backLeftSpeed) <= 0.01 && Math.abs(backRightSpeed) <= 0.01) {
-        frontLeftAngle = frontLeftPrevAngle;
-        frontRightAngle = frontRightPrevAngle;
-        backLeftAngle = backLeftPrevAngle;
-        backRightAngle = backRightPrevAngle;
-        }
+        // if (Math.abs(frontLeftSpeed) <= 0.01 && Math.abs(frontRightSpeed) <= 0.01
+        // && Math.abs(backLeftSpeed) <= 0.01 && Math.abs(backRightSpeed) <= 0.01) {
+        // frontLeftAngle = frontLeftPrevAngle;
+        // frontRightAngle = frontRightPrevAngle;
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java]: why is this being commented out in this change? If we actually want to get rid of the code, just get rid of the code. But this is another change that doesn't seem related to advantagekit and I don't understand why it's here

File: subsystems/drivebase/SwerveDrive.java [hunk 27]
```diff
@@ -288,28 +330,36 @@ public void initCoastMode() {
 
     public void toggleIdleMode() {
         for (SwerveModule m : mModules) {
-            if (m.getDriveMotor().getIdleMode() == IdleMode.kBrake)
-                    m.getDriveMotor().setIdleMode(IdleMode.kCoast);
+            if (((CANSparkMax) m.getDriveMotor()).getIdleMode() != IdleMode.kBrake)
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java]: I don't know why we're touching this in this change but it'll break with the kraken implementation

File: subsystems/drivebase/SwerveDrive.java [hunk 28]
```diff
@@ -288,28 +330,36 @@ public void initCoastMode() {
...
-                m.getDriveMotor().setIdleMode(IdleMode.kBrake);}
+                ((CANSparkMax) m.getDriveMotor()).setIdleMode(IdleMode.kBrake);}
             //We do not want to toggle steer motor idle mode 
         }
     }
 
     public void setMotorIdleMode(IdleMode mode) {
         for (SwerveModule m : mModules) {
-            (m.getDriveMotor()).setIdleMode(mode);
+            ((CANSparkMax) m.getDriveMotor()).setIdleMode(mode);
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java]: why this change? and it'll break with krakens

File: subsystems/drivebase/SwerveDrive.java [hunk 29]
```diff
@@ -288,28 +330,36 @@ public void initCoastMode() {
...
         }
     }
 
+    public void driveRobotRelative(ChassisSpeeds speeds) {
+        SwerveModuleState[] targetStates = kinematics.toSwerveModuleStates(speeds);
+
```
Comment (adias408) [inline subsystems/drivebase/SwerveDrive.java]: I'm not going to keep adding the "why are we changing this" comments in this file. Just assume that for all this stuff that has nothing to do with advantagekit

File: subsystems/drivebase/TankDrive.java [hunk 30]
```diff
@@ -592,8 +594,10 @@ public double getRightVelocityTicks() {
     return (right1Encoder.getVelocity() / RobotConstantsBase.TankDriveBase.kSparkTicksPerRotation / 60);
   }
 
-   //The driver never uses quickturn
-  public abstract boolean isQuickTurnPressed();
+   
+  public boolean isQuickTurnPressed() {
+    return OI.getDriverController().getRightBumper();
```
Comment (adias408) [inline subsystems/drivebase/TankDrive.java]: so confused. more stuff that has nothing to do with advantagekit

File: swervelib/DriveController.java [hunk 31]
```diff
@@ -1,9 +1,7 @@
 package frc.team670.mustanglib.swervelib;
 
-import com.revrobotics.CANSparkMax;
-
 public interface DriveController {
-    CANSparkMax getDriveMotor();
+    Object getDriveMotor();
```
Comment (adias408) [inline swervelib/DriveController.java]: ???

File: swervelib/DriveControllerFactory.java [hunk 32]
```diff
@@ -1,7 +1,7 @@
 package frc.team670.mustanglib.swervelib;
 
 import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardContainer;
-import frc.team670.mustanglib.utils.Logger;
+import frc.team670.mustanglib.utils.ConsoleLogger;
```
Comment (adias408) [inline swervelib/DriveControllerFactory.java]: it's not even used in this file, is it? if not, just delete

File: swervelib/Mk4ModuleConfiguration.java [hunk 33]
```diff
@@ -2,6 +2,8 @@
 
 import java.util.Objects;
 
+import frc.team670.mustanglib.swervelib.redux.AbsoluteEncoderType;
```
Comment (adias408) [inline swervelib/Mk4ModuleConfiguration.java]: why is this part of this change?

and also: it's weird that AbsoluteEncoderType is under a redux package - having it there already implies a type

File: swervelib/Mk4SwerveModuleBuilder.java [hunk 34]
```diff
@@ -55,7 +53,7 @@ private static SteerControllerFactory<?, SteerConfiguration<CanCoderAbsoluteConf
     private int steerMotorPort = -1;
     private String steerCanbus = "";
 
-
+    private MotorType steerMotorType;
```
Comment (adias408) [inline swervelib/Mk4SwerveModuleBuilder.java]: why is this here?

File: swervelib/Mk4iSwerveModuleHelper.java [hunk 35]
```diff
@@ -1,23 +1,25 @@
 package frc.team670.mustanglib.swervelib;
 
 import frc.team670.mustanglib.swervelib.ctre.*;
+import frc.team670.mustanglib.swervelib.redux.*;
```
Comment (adias408) [inline swervelib/Mk4iSwerveModuleHelper.java]: I'm going to stop reading here. I don't understand why all these unrelated changes are in here. Let's figure that out and then finish this

**Review by adias408 — CHANGES_REQUESTED** [review]
It's hard to read this. There are a lot of entirely unrelated changes in here. Do we understand why that is?

**Comment by arghunter** [comment]
> It's hard to read this. There are a lot of entirely unrelated changes in here. Do we understand why that is?
 
I merged the 2024-updated branch into this branch to reduce merge conflicts

File: swervelib/rev/NeoSteerControllerFactoryBuilder.java [hunk 36]
```diff
@@ -60,7 +60,7 @@ public FactoryImplementation(AbsoluteEncoderFactory<T> encoderFactory) {
 
         @Override
         public void addDashboardEntries(ShuffleboardContainer container, ControllerImplementation controller) {
-            // Logger.consoleLog("Called super.addDashboardEntries");
+            ConsoleLogger.consoleLog("Called super.addDashboardEntries");
```
Comment (adias408) [inline swervelib/rev/NeoSteerControllerFactoryBuilder.java]: why adding this back in? here and below

Comment (adias408) [reply swervelib/rev/NeoSteerControllerFactoryBuilder.java]: Actually below is just a rename - it's not adding it back. But why add it here?

**Review by adias408 — COMMENTED** [review]
Just a few comments. Rest looks fine

File: swervelib/rev/NeoSteerControllerFactoryBuilder.java (hunk 36 above)
Comment (arghunter) [reply swervelib/rev/NeoSteerControllerFactoryBuilder.java]: I'm not entirely sure why, I'll remove it

File: .vscode/settings.json [hunk 37]
```diff
@@ -0,0 +1,3 @@
+{
+    "workbench.iconTheme": "eq-material-theme-icons-darker"
```
Comment (adias408) [inline .vscode/settings.json]: is there a reason we're committing this file? seems like everyone could have different vscode settings

Comment (arghunter) [reply .vscode/settings.json]: i forgot to add it to the git ignore. I'll remove it.

**Comment by arghunter** [comment]
Advantagekit works and runs. I'm going to leave it running to test for packet loss but everything looks good so far. We still need a usb though

**Comment by adias408** [comment]
> Advantagekit works and runs. I'm going to leave it running to test for packet loss but everything looks good so far. We still need a usb though

Something like this? https://www.amazon.com/SanDisk-32GB-Ultra-Flash-Drive/dp/B09FFK1QLR/ref=sr_1_4

File: .vscode/settings.json (hunk 37 above)
Comment (adias408) [reply .vscode/settings.json]: bump

File: commands/RamseteCommand.java (hunk 12 above)
Comment (adias408) [reply commands/RamseteCommand.java:165]: bump

File: commands/drive/teleop/swerve/XboxSwerveDrive.java (hunk 13 above)
Comment (adias408) [reply commands/drive/teleop/swerve/XboxSwerveDrive.java]: bump

File: commands/drive/teleop/swerve/XboxSwerveDrive.java (hunk 14 above)
Comment (adias408) [reply commands/drive/teleop/swerve/XboxSwerveDrive.java]: bump

File: dataCollection/sensors/ColorMatcher.java (hunk 18 above)
Comment (adias408) [reply dataCollection/sensors/ColorMatcher.java:68]: bump

File: dataCollection/sensors/IRSensor.java (hunk 19 above)
Comment (adias408) [reply dataCollection/sensors/IRSensor.java]: bump

File: dataCollection/sensors/PicoColorMatcher.java (hunk 21 above)
Comment (adias408) [reply dataCollection/sensors/PicoColorMatcher.java:65]: bump

File: dataCollection/sensors/TimeOfFlightSensor.java (hunk 22 above)
Comment (adias408) [reply dataCollection/sensors/TimeOfFlightSensor.java]: bump

File: subsystems/MustangSubsystemBase.java (hunk 23 above)
Comment (adias408) [reply subsystems/MustangSubsystemBase.java]: bump

File: swervelib/DriveControllerFactory.java (hunk 32 above)
Comment (adias408) [reply swervelib/DriveControllerFactory.java]: bump

**Review by adias408 — CHANGES_REQUESTED** [review]

File: commands/RamseteCommand.java (hunk 12 above)
Comment (arghunter) [reply commands/RamseteCommand.java:165]: Yeah, we don't care much about them and we don't use Ramsete Command anymore. I'll leave these here for now and just delete the file in 2024-updated

File: commands/drive/teleop/swerve/XboxSwerveDrive.java (hunk 13 above)
Comment (arghunter) [reply commands/drive/teleop/swerve/XboxSwerveDrive.java]: so sebby wanted to slow down the swerve during drive practice one day so we added this and the comment below to allow for variable speed. I removed  it

File: commands/drive/teleop/swerve/XboxSwerveDrive.java (hunk 14 above)
Comment (arghunter) [reply commands/drive/teleop/swerve/XboxSwerveDrive.java]: Temporary speed control code. Removed. Look above for more details

File: dataCollection/sensors/ColorMatcher.java (hunk 18 above)
Comment (arghunter) [reply dataCollection/sensors/ColorMatcher.java:68]: That makes sense, although we should probably make that change on the 2024-updated branch

File: dataCollection/sensors/PicoColorMatcher.java (hunk 21 above)
Comment (arghunter) [reply dataCollection/sensors/PicoColorMatcher.java:65]: We can differentiate these. (Line 170 PicoColorSensor) int port = SerialPortJNI.serialInitializePort((byte)1);
We should be able to differentiate by passing a port in to the methdo

Comment (arghunter) [reply dataCollection/sensors/PicoColorMatcher.java:65]: Also, I'll make this change on 2024-updated

File: dataCollection/sensors/TimeOfFlightSensor.java (hunk 22 above)
Comment (arghunter) [reply dataCollection/sensors/TimeOfFlightSensor.java]: we don't

File: dataCollection/sensors/ColorMatcher.java (hunk 18 above)
Comment (adias408) [reply dataCollection/sensors/ColorMatcher.java:68]: add a trello card

File: dataCollection/sensors/PicoColorMatcher.java (hunk 21 above)
Comment (adias408) [reply dataCollection/sensors/PicoColorMatcher.java:65]: Ok. Add a trello card so we don't forget

File: utils/SwervePoseEstimatorBase.java (hunk 9 above)
Comment (adias408) [reply utils/SwervePoseEstimatorBase.java:23]: still wondering about this one - just delete the comment or make it an actual comment if there's actually a reason for it

**Review by adias408 — APPROVED** [review]
lgtm. there are still some things that ought to get cleaned up but as long as we don't lose track of them, it's fine
